tools/challenge-helper-scripts/utils.js

```diff
@@ -2,42 +2,86 @@ const fs = require('fs');
 const path = require('path');
 const matter = require('gray-matter');
 const ObjectID = require('bson-objectid');
+const { parseMDSync } = require('../challenge-parser/parser');
 
 const padWithLeadingZeros = originalNum => {
   /* always want file step numbers 3 digits */
   return ('' + originalNum).padStart(3, '0');
 };
 
-const removeErms = seedCode => {
+const insertErms = (seedCode, erms) => {
   const lines = seedCode.split('\n');
-  return lines
-    .filter(line => !line.includes('--fcc-editable-region--'))
-    .join('\n');
+  if (Number.isInteger(erms[0])) {
+    lines.splice(erms[0], 0, '--fcc-editable-region--');
+  }
+  if (Number.isInteger(erms[1])) {
+    lines.splice(erms[1], 0, '--fcc-editable-region--');
+  }
+  return lines.join('\n');
 };
 
 const createStepFile = ({
   projectPath,
   stepNum,
-  challengeSeed = '',
+  challengeSeeds = {},
   stepBetween = false
 }) => {
-  if (challengeSeed) {
-    challengeSeed = removeErms(challengeSeed);
-  }
+  const seedTexts = Object.values(challengeSeeds).map(
+    ({ contents, ext, editableRegionBoundaries }) => {
+      const fullContents = insertErms(contents, editableRegionBoundaries);
+      return `\`\`\`${ext}
+${fullContents}
+\`\`\``;
+    }
+  );
+
+  const seedHeads = Object.values(challengeSeeds)
+    .filter(({ head }) => head)
+    .map(
+      ({ ext, head }) => `\`\`\`${ext}
+${head}
+\`\`\``
+    )
+    .join('\n');
+
+  const seedTails = Object.values(challengeSeeds)
+    .filter(({ tail }) => tail)
+    .map(
+      ({ ext, tail }) => `\`\`\`${ext}
+${tail}
+\`\`\``
+    )
+    .join('\n');
 
   const descStepNum = stepBetween ? stepNum + 1 : stepNum;
   const stepDescription = `${
     stepBetween ? 'new' : ''
   } step ${descStepNum} instructions`;
-  const challengeSeedSection = `# --seed--
+  const challengeSeedSection = `
+# --seed--
 
-  ## --seed-contents--
+## --seed-contents--
 
-${challengeSeed.trim()}
+${seedTexts.join('\n')}`;
 
-`;
+  const seedHeadSection = seedHeads
+    ? `
 
-  const template = `---
+## --before-user-code--
+
+${seedHeads}`
+    : '';
+
+  const seedTailSection = seedTails
+    ? `
+
+## --after-user-code--
+
+${seedTails}`
+    : '';
```

Comment:
@ojeytonwilliams I don't think these lines are needed.  If before/after code is needed, the author of the step can add those later.

Reply:
True, but my rationale was that those really are part of the seed (it often won't work without them), so it seemed better to include them by default.  Is that not quite right?

Reply:
Each step is like our current challenges.  We only add those sections when they are needed for the tests of a particular challenge.  The same would go for the before/after sections of a step.

Reply:
Randy and I discussed this.  There's an argument for including these or not, but for now we're going to include them and see how it works in practice.